From f4dd93e9bdc732e4da4231a7a76c58363684199e Mon Sep 17 00:00:00 2001
From: Lenny Szubowicz <lszubowi@redhat.com>
Date: Tue, 9 Dec 2014 17:55:57 -0500
Subject: [redhat] kernel.spec: provide beta Red Hat CA cert on x86_64

Message-id: <1418147757-43375-1-git-send-email-lszubowi@redhat.com>
Patchwork-id: 101203
O-Subject: [RHEL 7.1 PATCH V2 BZ 1170551] [redhat] Provide Red Hat CA cert via kernel-doc rpm
Bugzilla: 1170551
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1170551

Description:
    Provide the Red Hat CA public key that authenticates the
    RHEL kernel in the kernel-doc rpm. Authentication of the
    kernel is enforced when UEFI Secure Boot is enabled.

    For a RHEL GA release, that public key is embedded in shim.efi
    and is made available automatically on each boot.

    However, for a Beta release, a Beta-specific private key is
    used to sign the kernel and the corresonding public Beta key
    is intentionally not embedded in shim.efi. Therefore the
    kernel-doc rpm puts the Beta public key certificate file in:
    /usr/share/doc/kernel-keys/<kernel-ver>/kernel-signing-ca.cer

    In order to boot RHEL 7.1 Beta with UEFI Secure Boot the system
    owner needs to first manually enroll the Beta public key into
    the Machine Owner Key (MOK) list using the mokutil program.
    When Beta is complete, the system owner will be advised to
    remove this Beta key from the MOK list.

Upstream Status: Red Hat only

Brew: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=8351877

Tested:
    1. Installed the brew-built kernel-doc rpm.
    2. Observed that kernel-signing-ca.cer is present in the expected
       directory.
    3. mokutil --import \
          /usr/share/doc/kernel-keys/<kernel-ver>/kernel-signing-ca.cer
    4. Rebooted to complete enrollment of pending MOK request
    5. On reset after MOK enrollment, enabled Secure Boot
       - RHEL 7.1 build kernel-3.10.0-212.el7.x86_64 authenticates

Signed-off-by: Lenny Szubowicz <lszubowi@redhat.com>
Signed-off-by: Peter Jones <pjones@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/kernel.spec.template b/redhat/kernel.spec.template
index 5fa4b78..0abf68b 100644
--- a/redhat/kernel.spec.template
+++ b/redhat/kernel.spec.template
@@ -1225,6 +1225,11 @@ popd
 make %{?cross_opts} DESTDIR=$RPM_BUILD_ROOT bootwrapper_install WRAPPER_OBJDIR=%{_libdir}/kernel-wrapper WRAPPER_DTSDIR=%{_libdir}/kernel-wrapper/dts
 %endif
 
+%if %{with_doc}
+# Red Hat UEFI Secure Boot CA cert, which can be used to authenticate the kernel
+mkdir -p $RPM_BUILD_ROOT%{_datadir}/doc/kernel-keys/%{rpmversion}-%{pkgrelease}
+install -m 0644 %{SOURCE13} $RPM_BUILD_ROOT%{_datadir}/doc/kernel-keys/%{rpmversion}-%{pkgrelease}/kernel-signing-ca.cer
+%endif
 
 ###
 ### clean
@@ -1366,6 +1371,9 @@ fi
 %dir %{_datadir}/doc/kernel-doc-%{rpmversion}/Documentation
 %dir %{_datadir}/doc/kernel-doc-%{rpmversion}
 %{_datadir}/man/man9/*
+%{_datadir}/doc/kernel-keys/%{rpmversion}-%{pkgrelease}/kernel-signing-ca.cer
+%dir %{_datadir}/doc/kernel-keys/%{rpmversion}-%{pkgrelease}
+%dir %{_datadir}/doc/kernel-keys
 %endif
 
 %if %{with_kernel_abi_whitelists}
-- 
1.7.1